src/DataBoxEdge/DataBoxEdge/Common/Cmdlets/Triggers/DataBoxEdgeTriggerNewCmdletBase.cs

```diff
@@ -0,0 +1,250 @@
+            ParameterSetName = PeriodicTimerTriggerResourceIdParameterSet,
+            HelpMessage = HelpMessageTrigger.PeriodicTimerEventStartTimeHelpMessage)]
+        public DateTime StartTime;
+
+        [Parameter(Mandatory = true,
+            ParameterSetName = PeriodicTimerTriggerParameterSet,
+            HelpMessage = HelpMessageTrigger.PeriodicTimerEventTopicHelpMessage)]
+        [Parameter(Mandatory = true,
+            ParameterSetName = PeriodicTimerTriggerResourceIdParameterSet,
+            HelpMessage = HelpMessageTrigger.PeriodicTimerEventTopicHelpMessage)]
+        public string Topic;
+
+        [Parameter(Mandatory = false, HelpMessage = Constants.AsJobHelpMessage)]
+        public SwitchParameter AsJob { get; set; }
+
+        private ResourceModel _trigger;
+
+        private ResourceModel GetResourceModel()
+        {
+            return TriggersOperationsExtensions.Get(
+                this.DataBoxEdgeManagementClient.Triggers,
+                this.DeviceName,
+                this.Name,
+                this.ResourceGroupName);
+        }
+
+        private string GetRoleId()
+        {
+            return RolesOperationsExtensions.Get(
+                this.DataBoxEdgeManagementClient.Roles,
+                this.DeviceName,
+                this.RoleName,
+                this.ResourceGroupName).Id;
+        }
+
+        private string GetShareId()
+        {
+            return SharesOperationsExtensions.Get(
+                this.DataBoxEdgeManagementClient.Shares,
+                this.DeviceName,
+                this.ShareName,
+                this.ResourceGroupName).Id;
+        }
+
+        private string GetResourceFoundMessage()
+        {
+            return string.Format("'{0}'{1}{2}'.",
+                HelpMessageTrigger.ObjectName, Constants.ResourceAlreadyExists, this.Name);
+        }
+
+        private bool DoesResourceExists()
+        {
+            try
+            {
+                var resource = GetResourceModel();
+                if (resource == null) return false;
+                var msg = GetResourceFoundMessage();
+                throw new Exception(msg);
+            }
+            catch (CloudException e)
+            {
+                if (e.Response.StatusCode == HttpStatusCode.NotFound)
+                {
+                    return false;
+                }
+
+                throw;
+            }
+        }
+
+        private PSResourceModel CreateResourceModel()
+        {
+            var roleSinkInfo = new RoleSinkInfo(this.RoleId);
+            if (this.FileEvent.IsPresent)
+            {
+                var fileSourceInfo = new FileSourceInfo(this.ShareId);
+                _trigger = new FileEventTrigger(fileSourceInfo, roleSinkInfo, name: this.Name);
+            }
+            else if (this.PeriodicTimerEvent.IsPresent)
+            {
+                var _periodicSourceInfo = new PeriodicTimerSourceInfo(this.StartTime, this.Schedule, this.Topic);
```

Comment:
I think it's better to rename it `periodicSourceInfo`

Reply:
Okay